crates/trie/sparse/src/trie.rs

```diff
@@ -267,6 +219,19 @@ impl SparseTrie {
         let revealed = self.as_revealed_mut()?;
         Some((revealed.root(), revealed.take_updates()))
     }
+
+    /// Returns a [`SparseTrie::Blind`] based on this one. If this instance was revealed, or was
+    /// itself a `Blind` with a pre-allocated [`RevealedSparseTrie`], this will return
+    /// a `Blind` carrying a reset pre-allocated [`RevealedSparseTrie`].
+    pub fn blinded(self) -> Self {
```

Comment:
the method name is a bit confusing IMO, as it doesn't communicate that we reset the trie to re-use allocations. Maybe just `clear`?